File: src/Areas.Domain/LocalAreas/LocalAreaDetails.cs
using Areas.Domain.LocalAreas.Enums;

namespace Areas.Domain.LocalAreas;

public class LocalAreaDetails(
    Guid id,
    String name,
    AreaType areaType,
    Int32 population,
    DateOnly establishmentDate,
    Double averageHotelBill,
    Boolean isHeroCity,
    String regionName
)
{
    public Guid Id { get; } = id;
    public String Name { get; } = name;
    public AreaType AreaType { get; } = areaType;
    public Int32 Population { get; } = population;
    public DateOnly EstablishmentDate { get; } = establishmentDate;
    public Double AverageHotelBill { get; } = averageHotelBill;
    public Boolean IsHeroCity { get; } = isHeroCity;
    public String RegionName{ get; } = regionName;
}

File: src/Areas.BackOffice/client-app/src/domain/localAreas/localAreaDetails.ts
import { Page } from "../../tools/types/page";
import { AreaType } from "./enums/areaType";

export class LocalAreaDetails {
  constructor(
    public readonly id: string,
    public readonly name: string,
    public readonly areaType: AreaType,
    public readonly population: number,
    public readonly establishmentDate: Date,
    public readonly averageHotelBill: number,
    public readonly isHeroCity: boolean,
    public readonly regionName: string,
  ) {}
}

export function mapToLocalAreaDetailsPage(data: any): Page<LocalAreaDetails> {
  return Page.convert(data, mapToLocalAreaDetails);
}

export function mapToLocalAreaDetailsList(data: any[]): LocalAreaDetails[] {
  return data.map(mapToLocalAreaDetails);
}

export function mapToLocalAreaDetails(data: any): LocalAreaDetails {
  return new LocalAreaDetails(
    data.id,
    data.name,
    data.areaType,
    data.population,
    data.establishmentDate,
    data.averageHotelBill,
    data.isHeroCity,
    data.regionName,
  );
}
